influxdb/influxdb.go

```diff
@@ -0,0 +1,43 @@
+package influxdb
+
+import (
+	"context"
+	"os"
+	"time"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go"
+)
+
+func WriteData(serviceName string, containerID string, hostName string, servicePort int, serviceIP string, serviceStatus string, serviceTags []string) {
+
+	bucketName := os.Getenv("bucket")
+	influxToken := os.Getenv("influx_token")
+	orgName := os.Getenv("org_name")
+	influxURL := os.Getenv("influx_url")
```

Comment:
Putting this access here, means that every time we want to write to InfluxDB, we need to query the environment variable from the system. Perhaps, we can create a `New()` function that will compose a struct where all this value is already defined and the `WriteData()` be a method of this structure. Then, when the software initialises we have something like:

```go
influx = influxdb.New()
```

The "client" structure can be something like:

```go
type InfluxDBClient struct {
    bucketName  string
    token              string
    orgName        string
    influxdbURL   string
}
```

The New() function can be something like:

```go
func New() InfluxDBClient {
    return InfluxDBClient{
        bucketName := os.Getenv("influx_token")
        .....
    }
}
```

and later, together with the `Metric` data structure, the `WriteData()` will be come a method and:

```go
func (c *InfluxDBClient) WriteData(m *metric) {
     /* write data to influxdb code goes here */
}

influx.WriteData(metric)
```

---
Given that we are going to init a client of the `influxdb` only once, we won't be querying the environment variables every time we want to write a datapoint into the database. Thus, the codepath will be a little optimize. 

With this structure, it allows us to also declare the `influxdb client` that is provided by the official InfluxDB client inside our `type InfluxDBClient struct` which also init the connection once when the system is start-up and re-use to write metrics. Creating the connection is expensive to the system, so this will also optimise the codebase as well:

```go
type InfluxDBClient struct {
    bucketName  string
    token              string
    orgName        string
    influxdbURL   string

    // private client to the InfluxDB database that will be init
    // when the system starts. 
    client influxdb2.Client
}
```